iocontrol/raspi/digital.go

```diff
@@ -0,0 +1,106 @@
+package raspi
+
+import (
+	"strings"
+
+	io "github.com/macformula/hil/iocontrol"
+	"github.com/pkg/errors"
+	"periph.io/x/conn/v3/gpio"
+	"periph.io/x/conn/v3/gpio/gpioreg"
+	"periph.io/x/host/v3"
+)
+
+// DigitalPin is a digital input/output pin for the Raspberry Pi
+type DigitalPin struct {
+	pin Pin
+	dir io.Direction
+	lvl io.Level
+
+	gpio gpio.PinIO
+}
+
+// NewDigitalPin returns a new DigitalPin
+func NewDigitalPin(
+	pin Pin,
+	dir io.Direction,
+) *DigitalPin {
+	digitalPin := &DigitalPin{
+		pin: pin,
+		dir: dir,
+	}
+
+	return digitalPin
+}
+
+// Open initializes the host device drivers and acquires a gpio pin
+func (d *DigitalPin) Open() error {
+	if !d.pin.IsAPin() {
+		return errors.New("provided pin is reserved or not available")
+	}
+
+	if _, err := host.Init(); err != nil {
+		return err
```

Comment:
Maybe wrap errors like these with function names so we can stay consistent with other parts of the hil repo. Otherwise LGTM